ITMO_FS/embedded/MOS.py

```diff
@@ -94,61 +79,54 @@ def __init__(
         self.sampling = sampling
         self.k_neighbors = k_neighbors
 
-    def _fit(self, X, y, **kwargs):
-        """
-            Runs the MOS algorithm on the specified dataset.
+    def _fit(self, X, y):
+        """Run the MOS algorithm on the specified dataset.
 
-            Parameters
-            ----------
-            **kwargs
-            X : array-like, shape (n_samples, n_features)
-                The input samples.
-            y : array-like, shape (n_samples)
-                The classes for the samples.
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The input samples.
+        y : array-like, shape (n_samples,)
+            The classes for the samples.
 
-            Returns
-            ------
-            None
+        Returns
+        -------
+        None
         """
-
         if self.loss not in ['hinge', 'log']:
+            getLogger(__name__).error(
+                "Loss should be 'hinge' or 'log', %s was passed", self.loss)
             raise KeyError(
-                "Loss should be 'hinge' or 'log', %r was passed" %
-                self.loss)
+                "Loss should be 'hinge' or 'log', %s was passed" % self.loss)
 
         if self.sampling:
             try:
                 X, y = SMOTE(
                     random_state=self.seed,
-                    k_neighbors=self.k_neighbors).fit_resample(
-                    X, y)
-            except ValueError as e:
-                print(
-                    'Couldn\'t perform SMOTE because n_neighbors is bigger '
-                    'than amount of instances in one of the classes.')
-                raise e
+                    k_neighbors=self.k_neighbors).fit_resample(X, y)
+            except ValueError:
+                getLogger(__name__).warning(
```

Comment:
add logger into beggining as global value

Reply:
might do something with moving all loggers to their respective module-wide __init__ files later